src/service/index.ts

```diff
@@ -5,35 +5,48 @@ import { EngineService } from "./engine";
 import { AccountService } from "./account";
 import { Authenticator } from "../auth";
 import { QueryFormatter } from "../formatter";
-import { AuthOptions, Context } from "../types";
-import { authDeprecationWarning } from "../common/errors";
+import { ConnectionOptions, Context } from "../types";
+import { Connection } from "../connection";
+import { FireboltCore } from "../core";
 
 export class ResourceManager {
   private context: Context;
+  accountName!: string;
   engine: EngineService;
   database: DatabaseService;
   account: AccountService;
 
-  constructor(context: {
-    httpClient: HttpClientInterface;
-    logger: LoggerInterface;
-    queryFormatter: QueryFormatter;
-    apiEndpoint: string;
-  }) {
-    this.context = {
-      ...context,
-      resourceManager: this
-    };
+  constructor(
+    context: {
+      httpClient: HttpClientInterface;
+      logger: LoggerInterface;
+      queryFormatter: QueryFormatter;
+      apiEndpoint: string;
+    },
+    connection: Connection | undefined = undefined
```

Comment:
my suggestion to put connection in context as well. WDYT?

Reply:
I think the only reason I have it as a separate parameter outside of context is so I don't have to create another type `ResourceMangerContext` that only differs from original `Context` by one field. I don't feel too strongly about it though, if the best practice is to add it in the context I'll refactor the code.